src/main/java/com/onlinebookstore/onlinebookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package com.onlinebookstore.onlinebookstore.service.impl;
+
+import com.onlinebookstore.onlinebookstore.dto.book.BookDtoWithoutCategoriesIds;
+import com.onlinebookstore.onlinebookstore.dto.category.CategoryRequestDto;
+import com.onlinebookstore.onlinebookstore.dto.category.CategoryResponseDto;
+import com.onlinebookstore.onlinebookstore.exeption.EntityNotFoundException;
+import com.onlinebookstore.onlinebookstore.mapper.BookMapper;
+import com.onlinebookstore.onlinebookstore.mapper.CategoryMapper;
+import com.onlinebookstore.onlinebookstore.model.Category;
+import com.onlinebookstore.onlinebookstore.repository.BookRepository;
+import com.onlinebookstore.onlinebookstore.repository.CategoryRepository;
+import com.onlinebookstore.onlinebookstore.service.interfaces.CategoryService;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final BookRepository bookRepository;
+    private final CategoryMapper categoryMapper;
+    private final BookMapper bookMapper;
+
+    @Override
+    public CategoryResponseDto getById(Long id) {
+        Category category = findByIdOrThrowException(id);
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public List<CategoryResponseDto> findAll(String string, Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<BookDtoWithoutCategoriesIds> getBooksByCategoryId(Long id, Pageable pageable) {

```

Comment:
There is inconsistency in method name some of them named: deleteCategoryById, updateCategoryById etc (with word 'Category' in name) some of them named getById, findAll (without word 'Category' in name). Please decide approach and use it everywhere, better to avoid name of entity in method name, because it is clear without additional words